cmd/utils/flags.go

```diff
@@ -123,3 +123,17 @@ var (
 		Usage: "Specify account type as secp256k1",
 	}
 )
+
+// FixFlagOrder allow us to use various flag order formats, eg: (gossamer init --config config.toml and  gossamer --config config.toml init)
+func FixFlagOrder(f func(ctx *cli.Context) error) func(*cli.Context) error {
+	return func(ctx *cli.Context) error {
+		for _, flagName := range ctx.FlagNames() {
+			if ctx.IsSet(flagName) {
+				if err := ctx.GlobalSet(flagName, ctx.String(flagName)); err != nil {
+					log.Error("Error when fixing flag", "flagName", flagName)
+				}
+			}
+		}
+		return f(ctx)
+	}
+}
```

Comment:
This is not how flags should be used. Global flags apply to `gossamer`, local flags should be applied to sub-commands (eg. `accounts`). This seems to be standard for most CLIs

Reply:
Well, the issue was reported as `./build/bin/gossamer init --genesis genesis.json` does not work.
Indeed, it does not work because its order is improper. It would work if used like: `./build/bin/gossamer --genesis genesis.json init`
So, the fix is needed to avoid issues like this.

Reply:
`--genesis` only applies to the `init` command, `gossamer --genesis genesis.json` shouldn't be allowed as the flag isn't relevant.

Reply:
Well, the issue is like I said: 
`gossamer init --genesis genesis.json` does not parse the flags correctly, and values are nil.
When I run `gossamer --genesis genesis.json init` all works well and flags are processed.

Reply:
I believe this issue lies in the usage of `context.GlobalBool` and `context.Bool`. Afaik, the first will only return flags on the main command (`gossamer`) and the other on the subcommand (`init`)

Reply:
(bool is just an example, applies for all flag types)

Reply:
Right, IMHO having the method FixFlagOrder executed on all cmds reduces the complexity and avoid us having to fight with how the flags all the time. AFAIK It does not add any collateral effects as well.